torch/testing/_internal/common_utils.py

```diff
@@ -2709,6 +2714,21 @@ def generate_values(base, densesize):
                   [[1, 0],
                    [0, 0]]], [(1, 1)], ([()] if enable_batch_variable_nse else []))]
 
+        def non_contiguous_copy(t, dim=-1, offset=0):
+            # return a copy of t that is non-contiguous along the
+            # given dimension and with the given storage offset
+            self.assertTrue(t.is_contiguous())
+            if dim < 0:
+                dim = dim + t.ndim
+            assert dim >= 0 and dim < t.ndim
+            step = max(2, offset + 1)
+            tmp = torch.zeros((*t.shape[:dim], t.shape[dim] * step, *t.shape[dim + 1:]), dtype=t.dtype, device=t.device)
+            tmp[(*((slice(None),) * dim), slice(offset, None, step))].copy_(t)
```

Comment:
nit: `tmp[..., slice(...)]` should work as well.

Reply:
Thanks! I'll apply this in some follow-up.

Reply:
Using `Ellipsis` does not work because `offset::step` is not the last dimension.